frame/contracts/src/gas.rs

```diff
@@ -111,7 +111,7 @@ where
 
 		// NOTE that it is ok to allocate all available gas since it still ensured
 		// by `charge` that it doesn't reach zero.
-		if self.gas_left < amount {
+		if self.gas_left.any_lt(amount) {
```

Comment:
would be good if @athei checks this.

Reply:
That looks good. I imagine the gas meter will be used also size metering and then it needs to error out if any of those is exhausted. We could replace this by an `ensure!`, though.